TypeScript 4.8 compatibility

<!--
👋 Hi, thanks for filing an issue on eslint-plugin-expect-type! 💖
Please fill out all fields below to ensure your issue is addressed.

If your issue doesn't provide enough info to fully explain or reproduce your bug, it will be closed. 😦
-->

### 🐛 Bug Report

- `eslint-plugin-expect-type` version: 0.2.0
- ESLint version: 8.23.0
- Node version: 16.17.0

#### Actual Behavior

Bumping projects to TS v4.8 makes the plugin error

```
/Users/cedric/Code/ninjasquad/examples/node_modules/typescript/lib/typescript.js:167586
                fileExists: function (fileName) { return host.fileExists(fileName); },
                                                              ^

TypeError: Error while loading rule 'expect-type/expect': host.fileExists is not a function
```

#### Expected Behavior

It would be awesome to support the latest TS version.

#### Reproduction

It looks like bumping TS in the package.json of the plugin repo, and running `npm test` reproduces the issue:

```
 FAIL  tests/rules/twoslash.test.ts
  ● Test suite failed to run

    src/rules/expect.ts:507:3 - error TS2739: Type '{ getCompilationSettings: () => CompilerOptions; getCurrentDirectory: () => string; getDefaultLibFileName: () => string; getScriptFileNames: () => string[]; getScriptSnapshot: (name: string) => IScriptSnapshot; getScriptVersion: () => string; }' is missing the following properties from type 'LanguageServiceHost': readFile, fileExists

    507   return {
          ~~~~~~~~
    508     getCompilationSettings: () => program.getCompilerOptions(),
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ...
    513     getScriptVersion: () => '1',
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    514   };
        ~~~~
```

--- a/README.md
+++ b/README.md
@@ -120,7 +120,7 @@ const vector = {
   x: 3,
   y: 4,
 };
-vector
+vector;
 // ^? const vector: {
 //        x: number;
 //        y: number;

--- a/package.json
+++ b/package.json
@@ -75,6 +75,6 @@
     "prettier": "^2.6.0",
     "ts-jest": "^27.1.3",
     "tsutils": "^3.21.0",
-    "typescript": "^4.6.2"
+    "typescript": "^4.8.2"
   }
 }

--- a/src/rules/expect.ts
+++ b/src/rules/expect.ts
@@ -511,6 +511,8 @@ function getLanguageServiceHost(program: ts.Program): ts.LanguageServiceHost {
     getScriptFileNames: () => program.getSourceFiles().map((sourceFile) => sourceFile.fileName),
     getScriptSnapshot: (name) => ts.ScriptSnapshot.fromString(program.getSourceFile(name)?.text ?? ''),
     getScriptVersion: () => '1',
+    fileExists: (path) => !!program.getSourceFile(path),
+    readFile: (path) => program.getSourceFile(path)?.text ?? '',
   };
 }
 

